starlette/routing.py

```diff
@@ -28,6 +29,16 @@ class Match(Enum):
     FULL = 2
 
 
+def assert_args_is_only_route(func: typing.Callable) -> typing.Callable:
+    @functools.wraps(func)
+    def wrapper(self: BaseRoute, *args: str, **kwargs: str) -> URLPath:
+        assert len(args) > 0, "Missing route name as the first argument."
+        assert len(args) < 2, "Invalid positional argument passed."
+        return func(self, *args, **kwargs)
```

Comment:
Should raise TypeError instead.
("assert" might be a no-op)

Reply:
The decorator could be renamed to "verify_args_is_only_route" then.

Reply:
Done in 6bf9b0e.